tests/ui/rfcs/rfc-2396-target_feature-11/fn-ptr.rs

```diff
@@ -2,9 +2,23 @@
 
 #![feature(target_feature_11)]
 
+#[target_feature(enable = "avx")]
+fn foo_avx() {}
+
 #[target_feature(enable = "sse2")]
 fn foo() {}
 
+#[target_feature(enable = "sse2")]
+fn bar() {
+    let foo: fn() = foo; // this is OK, as we have the necessary target features.
```

Comment:
Also add a run-pass test where a target feature fn returns a function pointer to a target feature function and then code outside the unsafe code calls that pointer.